include/dgl/pcg32.h

```diff
@@ -0,0 +1,72 @@
+/*!
+ *   Copyright 2014 Melissa O'Neill <oneill@pcg-random.org>
```

Comment:
Is it legitimate to just copy like this?

Reply:
I didn't directly copy. I actually basically ported it to c++ but the operator()() is mostly the same, because that is the definition of pcg32 basically. Anybody who implements pcg32 implements operator()() in the same way. For the copyright, I didn't know if I should copyright it for myself or directly use the original author's, decided to do the latter.

Reply:
You can check the referenced file to see what I am talking about, https://github.com/imneme/pcg-c-basic/blob/master/pcg_basic.c